### PR TITLE
Update to use Spring Boot 2.x and Camel 3.5.0

### PR DESCRIPTION
**Pull request by oliverbrandt**
- Bumps Camel version to 3.5.0
- Bumps Spring Cloud Release to Hoxton.SR8.
- Fixes incompatibility problems and failing unit tests.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -7,12 +7,12 @@
     <groupId>com.playtika.sleuth</groupId>
     <artifactId>sleuth-camel</artifactId>
     <packaging>pom</packaging>
-    <version>1.0.2-SNAPSHOT</version>
+    <version>3.5.0-SNAPSHOT</version>
```
Comment (tdanylchuk) [inline pom.xml]: please dont change project version

File: pom.xml [hunk 2]
```diff
@@ -38,27 +38,27 @@
     </modules>
 
     <properties>
-        <maven.compiler.source>1.8</maven.compiler.source>
-        <maven.compiler.target>1.8</maven.compiler.target>
+        <maven.compiler.source>11</maven.compiler.source>
```
Comment (tdanylchuk) [inline pom.xml]: any reason why need to force 11 java, i would prefer stick to 8 if possible

File: sleuth-camel-core/pom.xml [hunk 3]
```diff
@@ -35,11 +35,17 @@
...
-            <groupId>org.apache.camel</groupId>
-            <artifactId>camel-spring-boot</artifactId>
+            <groupId>org.apache.camel.springboot</groupId>
+            <artifactId>camel-spring-boot-starter</artifactId>
             <scope>provided</scope>
         </dependency>
 
+        <dependency>
+            <groupId>org.apache.camel.springboot</groupId>
+            <artifactId>camel-direct-starter</artifactId>
```
Comment (tdanylchuk) [inline sleuth-camel-core/pom.xml:45]: what for this dependency?

**Review by tdanylchuk — CHANGES_REQUESTED** [review]

File: sleuth-camel-core/pom.xml (hunk 3 above)
Comment (oliverbrandt) [reply sleuth-camel-core/pom.xml:45]: This is due to the changed Camle packaging since 3.0.
The scope will be changed to test.

**Review by tdanylchuk — APPROVED** [review]

**Comment by tdanylchuk** [comment]
thank you for your contribution!
release ```2.0.0``` should be in maven repo within 24 hours

**Pull request merged**